flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumerThread.java

```diff
@@ -105,6 +117,8 @@ public void run() {
 
 					break;
 				} else {
+					Thread.sleep(idleMillisBetweenFetches);
```

Comment:
I'm not sure if its a good idea to introduce this waiting time by default.
Imagine we are consuming slowly because downstream operations are expensive. Then, we would introduce an artificial slowdown (in latency and throughput). I would suggest to:
- Set the value by default to 0
- Only call sleep if the idle time is > 0.